Set PYD_PLATFORM_TAG for armv7 targets too

mstorsjo opened the pull request:
This makes an armv7-mingw build of python actually find the native modules.

naveen521kk commented:
This needs changes to sysconfig/distutils too so that when building the extension modules are built with correct suffix. See https://github.com/msys2-contrib/cpython-mingw/pull/3 on what files to change.

mstorsjo commented:
> This needs changes to sysconfig/distutils too so that when building the extension modules are built with correct suffix. See #3 on what files to change.

Thanks! I added similar changes in `Lib/sysconfig.py` and `Lib/distutils/util.py`, and also added recognition of the architecture in `Python/getcompiler.c` (which was necessary for getting the right arch marker in `sys.version`).

Any other obvious place where I should add more detection for this combination?

naveen521kk commented:
> Any other obvious place where I should add more detection for this combination?

I don't think so. Thanks for the PR. Is it possible to test via Github Actions?

mstorsjo commented:
> > Any other obvious place where I should add more detection for this combination?
> 
> I don't think so. Thanks for the PR. Is it possible to test via Github Actions?

Not straight away I think. There doesn't seem to be any existing Github Actions coverage of the mingw/aarch64 case either (which does exist in the code as is today). I don't think msys2 provides an environment for mingw/armv7 (there's some experimental ones for aarch64 though). I build for this target with https://github.com/mstorsjo/llvm-mingw.

mstorsjo commented:
Thanks!

jeremyd2019 commented:
Thanks.  I added aarch64 because I was building it, but I decided to punt on armv7 because nobody seemed to really care about it.

mstorsjo commented:
> Thanks. I added aarch64 because I was building it, but I decided to punt on armv7 because nobody seemed to really care about it.

I build python for all 4 architectures in releases of llvm-mingw these days, as it’s required for integrating lldb into some IDEs. It builds fine for all 4 architectures (after some amount of fixes to libffi earlier this year), but I hadn’t actually tested the python build much, especially on armv7. But now it does seem to behave better.
